src/open_samus_returns_rando/files/templates/randomizerpowerup.lua

```diff
@@ -96,6 +97,34 @@ function RandomizerPowerup.MarkLocationCollected(actorOrName)
     Blackboard.SetProp(playerSection, propName, "b", true)
 end
 
+local AeionAbilityClouds = {
+    {"s000_surface", "LE_SpecialAbility_ScanningPulse"},
+    {"s028_area2c", "LE_SpecialAbility_EnergyShield"},
+    {"s030_area3", "LE_SpecialAbility_EnergyWave"},
+    {"s060_area6", "LE_SpecialAbility_PhaseDisplacement"},
+}
+
+function RandomizerPowerup.ActivateSpecialEnergy(actorOrName)
+    local name = actorOrName.sName
+    local cloud = "TG_SpecialEnergyCloud"
+    local trigger = cloud .. string.sub(name, 11)
+
+    -- Powerups
+    if string.sub(name, 0, 8) == "LE_Power" and trigger ~= nil then
+        SpecialEnergyCloud.ActivateSpecialEnergy(trigger)
+    -- The Super Missile actor name has a typo, so explicity activate its cloud
+    elseif name == "LE_PoweUp_SuperMissile" then
+        SpecialEnergyCloud.ActivateSpecialEnergy(cloud .. "_SuperMissile")
+    end
```

Comment:
If I understand correctly, the special case is just for the typo in `PowerUp`.
Fwiw, you could just find the last `_` and take the sub string from there.
Something like
```lua
if string.sub(name, 0, 7) == "LE_Powe" then
    local indexstart, indexend = string.find(name, "_", 4)
    local trigger = cloud .. string.sub(name, indexend)
    SpecialEnergyCloud.ActivateSpecialEnergy(trigger)
end
```